### PR TITLE
Added new functions in the npc_mov module and updated relevant tests

### PR DESCRIPTION
**Pull request by torresszy**
The two major functions left unfinished from last year, `move_npc_indefinite` and `auto_gen_movement` have been implemented under the `npc_move` module in this pull request. `move_npc_indefinite` will move an existing NPC with an indefinite move type from one room to another. `auto_gen_movement` will randomly generate a list of rooms and append the list to a given NPC's path. In addition, we have implemented two new functions, `get_npc_num_rooms` and `get_num_rooms`. `get_npc_num_rooms` takes a `npc_mov` struct and returns the number of rooms a NPC has in its list of rooms to visit. `get_num_rooms` is a helper function for `auto_gen_movement` that returns the total number of rooms in a given game. In addition, new tests are created for these functions in `test_npc_move.c`. One thing to be noted is that in both `move_npc_indefinite` and `auto_gen_movement`, we have inherited the concept of time from last year NPC team's implementation. However, this concept is not yet well-established in Chiventure at this point. Therefore, Updates on these functions may be needed when a more matured system on "time" is developed in the future.

Summary of major changes:

Added new functions, `move_npc_indefinite`, `get_npc_num_rooms`, `get_num_rooms`,and `auto_gen_movement`, in `npc_move.c` and their headers in `npc_move.h`.
Added new tests for these functions in `test_npc_move.c`

### CODE REVIEW

File: tests/npc/test_npc_move.c [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+    cr_assert_str_eq(npc_mov->track,"room_to_add1",
+                    "move_npc_def() failed to move once");
+    
+    check2 = move_npc_definite(npc_mov);
+
+    cr_assert_str_eq(npc_mov->track,"room_to_add2",
+                    "move_npc_def() failed to move twice");
+}
+
```
Comment (mbutera17) [inline tests/npc/test_npc_move.c:192]: Include test for move_npc_indefinite from Xavier

File: tests/npc/test_rooms_npc.c [hunk 2]
```diff
@@ -135,6 +82,39 @@ Test (npcs_in_room, add_npc_to_room)
 }
 
 
+/* Tests delete_npc_from_room function */
+Test (npcs_in_room, delet_npc_from_room)
```
Comment (mbutera17) [inline tests/npc/test_rooms_npc.c:86]: Fix typo: "delet_npc_from_room" -> "delete_npc_from_room"

**Review by mbutera17 — CHANGES_REQUESTED** [review]
Looks great - Just need Xavier's addition and one small typo.

**Review by xlewellen — APPROVED** [review]
Everything looks good!

**Review by mbutera17 — APPROVED** [review]
Looks good! Changes requested in my previous review have been resolved.

File: tests/npc/test_npc_move.c [hunk 3]
```diff
@@ -184,3 +247,88 @@ Test(npc_mov, reverse_path)
...
+            cr_assert_str_eq(get_ldesc(curr_room), "room2 long long long",
+            "ldesc does not correspond");
+        } else if (!strncmp(curr_room->room_id, "room3", MAX_ID_LEN)) {
+            cr_assert_str_eq(get_ldesc(curr_room), "room3 long long long",
+            "ldesc does not correspond");
+        }
+    }
+
+    num_rooms_in_npc = get_npc_num_rooms(npc_mov);
+    cr_assert_eq(cnt, num_rooms_in_npc, "room_count returns %d, but there should be %d rooms in npc_mov",
```
Comment (lilyehsani) [inline tests/npc/test_npc_move.c:287]: Change the next couple of lines to be under 80 characters

File: tests/npc/test_npc_move.c [hunk 4]
```diff
@@ -184,3 +247,88 @@ Test(npc_mov, reverse_path)
...
```
Comment (lilyehsani) [inline tests/npc/test_npc_move.c:330]: Under 80 characters

**Review by lilyehsani — CHANGES_REQUESTED** [review]
Looks really good! Added a few comments, mostly style

File: src/npc/src/npc_move.c [hunk 5]
```diff
@@ -219,4 +244,92 @@ int move_npc_definite(npc_mov_t *npc_mov)
...
+	}
+
+    num_rooms = get_num_rooms(game);
+    num_rooms_to_add = (rand() % num_rooms) + 1;
+
+    for (int i = 0; i < num_rooms_to_add; i++) {
+            room_t *room_to_add = malloc(sizeof(room_t));
+
+            room_to_add = head->room;
+            head->room = head->next->room; // not sure if this is correct
```
[the remark this answers is not in the document]
Comment (mbutera17) [reply src/npc/src/npc_move.c:319]: Sorry about the confusion of this comment - I had initially written this when I was drafting the function to point out to Torres & Xavier where I was confused. Turns out my suspicion was correct and that line was fixed and now works as intended.

**Review by lilyehsani — APPROVED** [review]
LGTM! Approved.

**Comment by torresszy** [comment]
This is the closing comment for this pull request; it has been merged successfully and all goals have been achieved. No major changes were needed since the beginning of this pull request. There were a few formatting and stylistic issues but they were resolved very quickly.

One note for future implementation, as I mentioned at the beginning of this PR as well, is that `move_npc_indefinite` and `auto_gen_movement` under `npc_move.h` inherit mainly the implementation and design logic from last year's NPC team, which includes this concept of "time" in seconds. However, This feature is currently not supported by Chiventure, so these two functions cannot be used practically at the moment unless with some modifications. Future teams can and should think about 1) how to bring the concept of "time" into the overall Chiventure, or 2) another way to keep track of the progression of the game.

**Comment by lilyehsani** [comment]
Issue Score: ✔️++

Comments:
Great work overall! The summary and closing of this PR are very thorough.

**Pull request merged**